Development repo spotlight

Rodriguez-Carlos-BAH opened the pull request:
- Implementation of carousel of cards component.
- Home page changed to display carousel of cards.
- Card component
  - Refactored to show/hide the projects stats section.
  - Refactored to display or not the Featured ribbon decorator base on property and data.
  - Adjust unit test to changes.
- Delete of Card-Featured component
- Code quality
  - Consolidation of Dialogs code into a Class DialogFunctions.
  - Alignment of unit test for changes related to Dialogs code consolidation.
- Version set to 3.0
- Fixes
  - Remove usage of virtual-repeat due a problem on refreshing after a dialog is open and close.
  - Dialog "Are you using this project?" from project details - dependencies

mvs5465 committed:
Merge pull request #101 from usdot-jpo-codehub/development

Development

Rodriguez-Carlos-BAH committed:
Merge pull request #108 from usdot-jpo-codehub/development

Code promotion

Rodriguez-Carlos-BAH committed:
Merge pull request #110 from usdot-jpo-codehub/development

code promotion

Rodriguez-Carlos-BAH committed:
Merge pull request #112 from usdot-jpo-codehub/development

Code promotion

mary-vandyke committed:
Merge pull request #117 from usdot-jpo-codehub/development

Development

Rodriguez-Carlos-BAH committed:
RDA-1509 : Repository Spotlight implementation

- Implementation of carousel of cards component.
  - Unit test added

- Home page changed to display carousel of cards.

- Card component
  - Refactored to show/hide the projects stats section.
  - Refactored to display or not the Featured ribbon decorator base on property and data.
  - Adjust unit test to changes.

- Delete of Card-Featured component
  - This componente is replaced by the regular Card component.
  - Remove related unit test

- Code quality
  - Consolidation of Dialogs code into a Class DialogFunctions.
  - Alignment of unit test for changes related to Dialogs code consolidation.

- Version set to 3.0

- Fixes
  - Remove usage of virtual-repeat due a problem on refreshing after a dialog is open and close.
    - This problem needs to be solve through the implementation of pagination.
  - Dialog "Are you using this project?" from project details - dependencies
    - The data for the list of projects was not populated correctly.

Rodriguez-Carlos-BAH committed:
RDA-1509 : README update.

Rodriguez-Carlos-BAH committed:
Merge branch 'master' into development_repo-spotlight

mvs5465 commented on src/components/modals/addprojects-modal.js:
```diff
@@ -24,7 +24,7 @@ export class AddProjectsModal {
     return this.dataContext.getRepositories(null)
       .then(projects => {
-          this.projects = JSON.parse(JSON.stringify(projects));
+          // this.projects = JSON.parse(JSON.stringify(projects));
```
Would be helpful to note why this is commented out, or remove it.

Rodriguez-Carlos-BAH replied:
good catch... fixed!

mvs5465 reviewed (approved):
Great work with the DRY principles. Moving the dialog-functions into their own file looks like a huge headache relief. 

Nice work!

Rodriguez-Carlos-BAH committed:
RDA-1509 : Remove commented code.

Rodriguez-Carlos-BAH committed:
Merge branch 'development_repo-spotlight' of github.com:usdot-jpo-codehub/codehub-ui into development_repo-spotlight

Rodriguez-Carlos-BAH committed:
RDA-1509 : Synch changes with "development" branch.

Rodriguez-Carlos-BAH committed:
RDA-1543 : Getting on synch with Development

Rodriguez-Carlos-BAH committed:
RDA-1501 : Popular Categories

- Alignment with Repository Spotlight feature
- Integration of Popular Categories with Explore view
- Remove not needed console.log
- Unit test for Popular Categories and Category Card components
- Unit test for Explorer integration
- Update Repository Card component
  - Exchange places between Source and Owner
  - Update Unit Test

Rodriguez-Carlos-BAH committed:
RDA-1501 : Adjusting styles.

Rodriguez-Carlos-BAH committed:
Home Header text
- Change: ITS software code. -> ITS source code.
- Change: Remove the word "public" from: ...Transportation’s public portal for open-source...

Rodriguez-Carlos-BAH committed:
RDA-1501 : Align unit test.
